iothub/device/src/IotHubConnectionString.Core.cs

```diff
@@ -24,9 +24,17 @@ Task<string> IAuthorizationProvider.GetPasswordAsync()
                         || TokenRefresher != null,
                     "The token refresher and the shared access signature can't both be null");
 
-                return !string.IsNullOrWhiteSpace(SharedAccessSignature)
```

Comment:
This logic was faulty, and produced the NRE. By making the method `async`, we return the value, not the Task. This way the caller will get a Task with the result of `null` instead of an NRE when evaluating the Task.

It shouldn't happen, but it did, so better have code that handles it.